src/dump.ts

```diff
@@ -59,4 +63,37 @@ namespace Il2Cpp {
         file.close();
         ok(`dump saved to ${destination}`);
     }
+
+    export function dumpTree(path?: string): void {
+        const basePath = path ?? `${Il2Cpp.application.dataPath!}/${Il2Cpp.application.identifier ?? "unknown"}_${Il2Cpp.application.version ?? "unknown"}`;
+        const basePathExists = Il2Cpp.corlib.class("System.IO.Directory").method<boolean>("Exists").invoke(Il2Cpp.string(basePath));
+
+        if (basePathExists) {
+            warn(`directory ${basePath} already exists, skipping...`);
+            return;
+        }
```

Comment:
I would add a parameter (e.g. `ignoreAlreadyExistingDirectory: boolean = false`) to allow writing to an already existing folder. Please extract `directoryExists` to a function (just like `createDirectoryRecursively`):

```suggestion
    export function dumpTree(path?: string, ignoreAlreadyExistingDirectory: boolean = false): void {
        const basePath = path ?? `${Il2Cpp.application.dataPath!}/${Il2Cpp.application.identifier ?? "unknown"}_${Il2Cpp.application.version ?? "unknown"}`;

        if (!ignoreAlreadyExistingDirectory && directoryExists(basePath)) {
            raise(`directory ${basePath} already exists - pass ignoreAlreadyExistingDirectory = true to skip this check`);
        }
```

Reply:
Ah, not _that_ kind of inlining 👍